cmd/backup-manager/app/backup/backup.go

```diff
@@ -14,200 +14,128 @@
 package backup
 
 import (
-	"database/sql"
+	"context"
 	"fmt"
-	"io/ioutil"
+	"io"
 	"os/exec"
-	"path/filepath"
-	"strconv"
-	"strings"
-	"time"
 
-	"github.com/mholt/archiver"
-	"github.com/pingcap/tidb-operator/cmd/backup-manager/app/constants"
-	"github.com/pingcap/tidb-operator/cmd/backup-manager/app/util"
+	"github.com/gogo/protobuf/proto"
 	glog "k8s.io/klog"
-)
-
-// BackupOpts contains the input arguments to the backup command
-type BackupOpts struct {
-	Namespace   string
-	BackupName  string
-	Bucket      string
-	Host        string
-	Port        int32
-	Password    string
-	User        string
-	StorageType string
-}
 
-func (bo *BackupOpts) String() string {
-	return fmt.Sprintf("%s/%s", bo.Namespace, bo.BackupName)
-}
+	kvbackup "github.com/pingcap/kvproto/pkg/backup"
+	"github.com/pingcap/tidb-operator/cmd/backup-manager/app/util"
+	"github.com/pingcap/tidb-operator/pkg/apis/pingcap/v1alpha1"
+)
 
-func (bo *BackupOpts) getBackupFullPath() string {
-	return filepath.Join(constants.BackupRootPath, bo.getBackupRelativePath())
-}
+const (
+	// MetaFile represents file name
+	MetaFile = "backupmeta"
+)
 
-func (bo *BackupOpts) getBackupRelativePath() string {
-	backupName := fmt.Sprintf("backup-%s", time.Now().UTC().Format(time.RFC3339))
-	return fmt.Sprintf("%s/%s", bo.Bucket, backupName)
+// Options contains the input arguments to the backup command
+type Options struct {
+	Namespace  string
+	BackupName string
 }
 
-func (bo *BackupOpts) getDestBucketURI(remotePath string) string {
-	return fmt.Sprintf("%s://%s", bo.StorageType, remotePath)
+func (bo *Options) String() string {
+	return fmt.Sprintf("%s/%s", bo.Namespace, bo.BackupName)
 }
 
-func (bo *BackupOpts) getTikvGCLifeTime(db *sql.DB) (string, error) {
-	var tikvGCTime string
-	sql := fmt.Sprintf("select variable_value from %s where variable_name= ?", constants.TidbMetaTable)
-	row := db.QueryRow(sql, constants.TikvGCVariable)
-	err := row.Scan(&tikvGCTime)
+func (bo *Options) backupData(backup *v1alpha1.Backup) (string, error) {
+	args, path, err := constructOptions(backup)
 	if err != nil {
-		return tikvGCTime, fmt.Errorf("query cluster %s %s failed, sql: %s, err: %v", bo, constants.TikvGCVariable, sql, err)
+		return "", err
 	}
-	return tikvGCTime, nil
-}
-
-func (bo *BackupOpts) setTikvGCLifeTime(db *sql.DB, gcTime string) error {
-	sql := fmt.Sprintf("update %s set variable_value = ? where variable_name = ?", constants.TidbMetaTable)
-	_, err := db.Exec(sql, gcTime, constants.TikvGCVariable)
+	var btype string
+	if backup.Spec.Type == "" {
+		btype = string(v1alpha1.BackupTypeFull)
+	} else {
+		btype = string(backup.Spec.Type)
```

Comment:
BR needs `BackupType` as args?

Reply:
BR support `full` and `table` subcommand, but `BackupType` only contain `full` and `incremental`. So I think we need to add a new BackupType `table`. In addition, we need to check the three BackupTypes here. For br, only` full` and `table` BackupType is valid.

Reply:
I have added type `db` and `table` for br [here](https://github.com/pingcap/tidb-operator/pull/1280/files#diff-98e6ad9bd9688c018746d6bf46e0e6d6R596).
I will add validation in the [validBackup](https://github.com/pingcap/tidb-operator/pull/1280/files#diff-5d667fa465378b0536cbc88f73233d91R223) function.